src/gui/qgsfilterlineedit.cpp

```diff
@@ -28,84 +29,42 @@ QgsFilterLineEdit::QgsFilterLineEdit( QWidget *parent, const QString &nullValue
   : QLineEdit( parent )
   , mNullValue( nullValue )
 {
-  // need mouse tracking to handle cursor changes
-  setMouseTracking( true );
-
-  QIcon clearIcon = QgsApplication::getThemeIcon( "/mIconClearText.svg" );
-
   // icon size is about 2/3 height of text, but minimum size of 16
   int iconSize = std::floor( std::max( Qgis::UI_SCALE_FACTOR * fontMetrics().height() * 0.75, 16.0 ) );
 
-  mClearIconSize = QSize( iconSize, iconSize );
-  mClearIconPixmap = clearIcon.pixmap( mClearIconSize );
-  QIcon hoverIcon = QgsApplication::getThemeIcon( "/mIconClearTextHover.svg" );
-  mClearHoverPixmap = hoverIcon.pixmap( mClearIconSize );
+  QIcon clearIcon;
+  clearIcon.addPixmap( QgsApplication::getThemeIcon( "/mIconClearText.svg" ).pixmap( QSize( iconSize, iconSize ) ), QIcon::Selected, QIcon::On );
+  clearIcon.addPixmap( QgsApplication::getThemeIcon( "/mIconClearTextHover.svg" ).pixmap( QSize( iconSize, iconSize ) ), QIcon::Active, QIcon::On );
+  mClearAction = new QAction( clearIcon, tr( "clear" ), this );
```

Comment:
I'd change ```tr ("clear") )``` to just QString() - otherwise it adds an unwanted "clear" tooltip. Ideally this tooltip be adjustable so it could say "Reset to NULL", "Reset to default", etc. But an easier approach is just to remove the tooltip entirely.